### ISSUE TITLE
GrowTracker 3.0

### ISSUE THREAD

**Issue by 7LPdWcaW** [7LPdWcaW/GrowTracker-Android#206]
Read more: https://www.reddit.com/r/GrowUtils/comments/gjlrsg/growtracker_v27_prealpha_and_announcement/

GrowTracker is old. Very old. About 5 years old and the code is a hot mess of old hacks, new features, kotlin, and no proper architecture standards. The code is extremely out of date with the latest Android development practices, and so am I. I've decided, that during the current situation we all find ourselves in, I would use this opportunity to learn the newest practices and development techniques by re-writing GrowTracker from the ground-up. I will be re-structuring the app in a new way and introducing new features and I will attempt to keep some current features also.

I have also commissioned for some designs to be made up and some graphics created to try modernise the look of the app and make it more user friendly.

I dont expect this to be something I'll finish soon, like the current app, will be developed on over time. I will continue to maintain the current version and push out bug fixes, but no new major features will be introduced.

Thanks for using GrowTracker and I hope the new version will be even better and more intuitive than the current version

**Comment by 7LPdWcaW** [comment]
Just a brief update into the development status of GrowTracker.

TL;DR Project is still active, just not as quick as I would like. 

I still have plans to re-write the app and improve it, however, as we all know the coronavirus has taken a much bigger hold on countries than it should have and so development has been impacted because of that. I still want to finish it, if not for my own use, but finding both the time and motivation whilst self imprisoned has proven difficult.

Bug fixes and the like which are small and easy to fix will still happen, but large functionality which requires dedicated time is difficult to achieve at the moment.

Thanks for sticking by!

**Comment by masterofobzene** [comment]
Is this dead or still in development?

> self imprisoned

What do you mean? quarantine?  what a better time to code than that? 

Anyway, thanks for the efforts.

**Comment by 7LPdWcaW** [comment]
Hi @masterofobzene 

As it currently stands, the project is on a hiatus. v2.x (current production) is still in maintenance mode. The pandemic was difficult for me as I am a software developer by trade so effectively working 7 days a week was just not viable. I personally still use the app so I have a vested interest in keeping it alive, I am just unable to dedicate the time it needs right now. 

I am currently working on a fix for a bug in the current version but it requires a bit of work to do.

**Comment by Jojona1233** [comment]
Heyjo ;) Just wondering how it is going? Are you still planning on version 3? I really like the app...

**Comment by 7LPdWcaW** [comment]
> Heyjo ;) Just wondering how it is going? Are you still planning on version 3? I really like the app...

Hi @Jojona1233, It has been a long time since I last logged in to this account. To be completely honest, and as much as I dont want to say the project is dead, but I have no plans on working on this any further as I simply do not have the time required to finish building and then maintaining the app, as much as I would like. So this project is in hiatus until I am able to start contributing again (if I am able). Version 3 was still hundreds of hours of development time away from even a simple usable app.
